Source/ui_unix/GamePad/GamePadDeviceListener_OSX.cpp

```diff
@@ -0,0 +1,350 @@
+#include "GamePadDeviceListener_OSX.h"
+#include "GamePadUtils.h"
+
+CGamePadDeviceListener::CGamePadDeviceListener(OnInputEvent OnInputEventCallBack, bool filter)
+    : OnInputEventCallBack(OnInputEventCallBack)
+    , m_running(true)
+    , m_filter(filter)
+{
+	m_thread = std::thread([this]() { InputDeviceListenerThread(); });
+}
+
+CGamePadDeviceListener::CGamePadDeviceListener(bool filter)
+    : CGamePadDeviceListener(nullptr, filter)
+{
+}
+
+CGamePadDeviceListener::~CGamePadDeviceListener()
+{
+	m_running = false;
+	CFRunLoopStop(CFRunLoopGetCurrent());
```

Comment:
This doesn't seem necessary, the loop in the thread will exit because of m_running being set to false. I also think that might have unintended side effects since we're stopping the run-loop for the current thread that might be used for other system stuff. Should be removed.